moabb/datasets/download.py

```diff
@@ -51,15 +54,194 @@ def data_path(url, sign, path=None, force_update=False, update_path=True, verbos
     if get_config(key) is None:
         set_config(key, osp.join(osp.expanduser("~"), "mne_data"))
     path = _get_path(path, key, sign)
-    destination = _url_to_local_path(url, op.join(path, key_dest))
+    destination = _url_to_local_path(url, osp.join(path, key_dest))
     # Fetch the file
-    if not op.isfile(destination) or force_update:
-        if op.isfile(destination):
+    if not osp.isfile(destination) or force_update:
+        if osp.isfile(destination):
             os.remove(destination)
-        if not op.isdir(op.dirname(destination)):
-            os.makedirs(op.dirname(destination))
+        if not osp.isdir(osp.dirname(destination)):
+            os.makedirs(osp.dirname(destination))
         _fetch_file(url, destination, print_destination=False)
 
     # Offer to update the path
     _do_path_update(path, update_path, key, sign)
     return destination
+
+
+@verbose
+def data_dl(url, sign, path=None, force_update=False, update_path=True, verbose=None):
+    """Download file from url to specified path
+
+    This function should replace data_path as the MNE will not support the download
+    of dataset anymore. This version is using Pooch.
+    Beware that now, all the file are stored directly in
+
+    Parameters
+    ----------
+    url : str
+        Path to remote location of data
+    sign : str
+        Signifier of dataset
+    path : None | str
+        Location of where to look for the BNCI data storing location.
+        If None, the environment variable or config parameter
+        ``MNE_DATASETS_(signifier)_PATH`` is used. If it doesn't exist, the
+        "~/mne_data" directory is used. If the dataset
+        is not found under the given path, the data
+        will be automatically downloaded to the specified folder.
+    force_update : bool
+        Force update of the dataset even if a local copy exists.
+    update_path : bool | None
+        If True, set the MNE_DATASETS_(signifier)_PATH in mne-python
+        config to the given path. If None, the user is prompted.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see :func:`mne.verbose`).
+
+    Returns
+    -------
+    path : list of str
+        Local path to the given data file. This path is contained inside a list
+        of length one, for compatibility.
+    """
+    sign = sign.upper()
+    key = "MNE_DATASETS_{:s}_PATH".format(sign)
```

Comment:
For now I guess this is okay, maybe we should have some milestone where we unify string formatting, e.g. using f-strings?

Reply:
Yes, it could be nice to have a uniform handling of string. As there is not too much string meddling or that it is pretty localized, we could keep good old fashion strings for now.